lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
+    ## Either a value or an error.
+    case isSuccess*: bool:
+    of true:
+      value*: T
+    of false:
+      error*: ref Exception
+      backtrace*: BacktraceFrame
```

Comment:
Exceptions already support backtraces. And the implementation does not use a linked list of GC'ed objects. ;-)